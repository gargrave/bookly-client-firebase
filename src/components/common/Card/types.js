// @flow
export type CardTextListItem = {
  title?: string,
  value?: string,
}
