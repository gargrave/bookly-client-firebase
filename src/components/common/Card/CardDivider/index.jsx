// @flow
import React from 'react';

import { buildClasses } from '../../../../utils/cssHelpers';

type Props = {
};

const CardDivider = ({}: Props) => {
  return (
    <hr className={buildClasses(['card__divider'])} />
  );
};

CardDivider.propTypes = {
};

export default CardDivider;
