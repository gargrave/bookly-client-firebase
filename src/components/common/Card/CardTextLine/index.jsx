// @flow
import React from 'react';
import { bool, string } from 'prop-types';

import { buildClasses } from '../../../../utils/cssHelpers';

type Props = {
  bold?: boolean,
  condensed?: boolean,
  text: string,
};

const CardTextLine = ({
  bold = false,
  condensed = false,
  text,
}: Props) => {
  return (
    <p
      className={buildClasses([
        'card__text',
        bold ? 'card__text--bold' : '',
        condensed ? 'condensed' : '',
      ])}
    >
      {text}
    </p>
  );
};

CardTextLine.propTypes = {
  bold: bool,
  condensed: bool,
  text: string.isRequired,
};

export default CardTextLine;
