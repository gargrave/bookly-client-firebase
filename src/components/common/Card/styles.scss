@import '../../../styles/_index';

.bookly-card {
  @include boxShadowMed;
  background-color: #f9f9f9;
  border: solid 1px #bbb;
  border-radius: 5px;
  margin: 8px auto;
  max-width: 600px;
  padding: 10px 20px;

  &--top-margin-med {
    margin-top: 50px;
  }

  hr {
    margin: 12px 0;
  }

  &--hoverable:hover {
    background: #e4effc;
    cursor: pointer;
  }

  &__header,
  &__title,
  &__text {
    margin: 0;
  }

  &__header,
  &__title {
    font-weight: bold;
  }

  &__header {
    font-size: 1.5rem;
  }

  &__text {
    color: #6e6969;
  }

  &__text + &__text {
    margin-top: 8px;
  }
}

.bookly-detail-card {
  padding: 15px;

  .bookly-card__title {
    font-size: 1.3rem;
    line-height: 2.5rem;
  }

  .bookly-card__title + .bookly-card__text {
    font-size: 1.1rem;
  }
}

.bookly {
  &-list-detail-card {
    padding: 10px
  }
}
