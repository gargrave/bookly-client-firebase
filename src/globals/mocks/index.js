import { authorMocks } from './data/authorMocks';
import { bookBucketMocks } from './data/bookBucketMocks';
import { bookMocks } from './data/bookMocks';
import { profileMocks } from './data/profileMocks';
import { userMocks } from './data/userMocks';

export {
  authorMocks,
  bookBucketMocks,
  bookMocks,
  profileMocks,
  userMocks,
};
