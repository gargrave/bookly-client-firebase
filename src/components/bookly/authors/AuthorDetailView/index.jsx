// @flow
import React from 'react';
import { func, instanceOf, oneOfType, shape, string } from 'prop-types';
import { format } from 'date-fns';

import type { Author } from '../../../../constants/flowtypes';

import { buildClasses } from '../../../../utils/cssHelpers';

import Alert from '../../../common/Alert';
import Button from '../../../common/Button';
import ButtonRow from '../../../common/ButtonRow';
import Card from '../../../common/Card';

type Props = {
  author: Author,
  onBackClick: Function,
  onDeleteClick: Function,
  onEditClick: Function,
  topLevelError?: string,
};

function AuthorDetailView({
  author,
  onBackClick,
  onDeleteClick,
  onEditClick,
  topLevelError,
}: Props) {
  return (
    <div className={buildClasses('author-detail-view')}>
      {topLevelError &&
        <Alert
          message={topLevelError}
          type="danger"
        />
      }
      <Card
        classes={['card--top-margin-med', 'detail-card', 'author-detail-card']}
        hoverable={false}
        title={`${author.firstName} ${author.lastName}`}
      >

        <hr/>
        <p className={buildClasses('card-text')}>
          <strong>Added:</strong> {format(author.created, 'MMM. DD, YYYY, HH:mm:ss')}
        </p>
        <p className={buildClasses('card-text')}>
          <strong>Updated:</strong> {format(author.updated, 'MMM. DD, YYYY, HH:mm:ss')}
        </p>

        <hr/>

        <ButtonRow>
          <Button
            onClick={onEditClick}
            position="left"
            text="Edit"
            type="info"
          />
          <Button
            onClick={onDeleteClick}
            text="Delete"
            type="danger"
          />
          <Button
            onClick={onBackClick}
            position="right"
            text="Back"
            type="light"
          />
        </ButtonRow>
      </Card>
    </div>
  );
}

AuthorDetailView.propTypes = {
  author: shape({
    id: string,
    created: oneOfType([
      instanceOf(Date),
      string,
    ]),
    updated: oneOfType([
      instanceOf(Date),
      string,
    ]),
    firstName: string,
    lastName: string,
  }),
  onBackClick: func.isRequired,
  onDeleteClick: func.isRequired,
  onEditClick: func.isRequired,
  topLevelError: string,
};

export default AuthorDetailView;
