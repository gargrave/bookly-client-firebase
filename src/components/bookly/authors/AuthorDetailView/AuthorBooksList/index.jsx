// @flow
import React from 'react';
import { array, func, object } from 'prop-types';

import type { Author, Book } from '../../../../../constants/flowtypes';

import { buildClasses } from '../../../../../utils/cssHelpers';

import BookList from '../../../books/BookList';
import Button from '../../../../common/Button';

import './styles.css';

type Props = {
  author: Author,
  books: Book[],
  onBookAddClick: Function,
  onBookClick: Function,
};

const prefixClass = (cls = '') => `author-books-list${cls}`;

const AuthorBooksList = ({
  author,
  books,
  onBookAddClick,
  onBookClick,
}: Props) => {
  const authorName = `${author.firstName} ${author.lastName}`;

  return (
    <div className={buildClasses(prefixClass())}>
      <h5 className={buildClasses(prefixClass('__header'))}>
        Books by {authorName}
        <Button
          onClick={onBookAddClick.bind(this, author)}
          text="Add"
          type="success"
        />
      </h5>
      <BookList
        books={books}
        onBookClick={onBookClick}
      />
    </div>
  );
};

AuthorBooksList.propTypes = {
  author: object.isRequired,
  books: array.isRequired,
  onBookAddClick: func.isRequired,
  onBookClick: func.isRequired,
};

export default AuthorBooksList;
