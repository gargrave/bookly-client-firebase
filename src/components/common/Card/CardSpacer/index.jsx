// @flow
import React from 'react';
import { oneOf } from 'prop-types';

import { buildClasses } from '../../../../utils/cssHelpers';

import './styles.css';

type Props = {
  size?: string,
};

const CardSpacer = ({
  size = 'medium',
}: Props) => {
  return (
    <div className={buildClasses(`card__spacer--${size}`)} />
  );
};

CardSpacer.propTypes = {
  size: oneOf(['small', 'medium', 'large']),
};

export default CardSpacer;
