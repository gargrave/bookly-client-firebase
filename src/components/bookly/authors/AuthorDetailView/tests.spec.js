import React from 'react';
import { shallow } from 'enzyme';

import { authorMocks } from '../../../../utils/mocks/';

import Card from '../../../common/Card';

import AuthorDetailView from './';

describe('AuthorDetailView', () => {
  let props;
  let component;

  beforeEach(() => {
    props = {
      author: Object.create(authorMocks[0]),
      onBackClick: jest.fn(),
      onDeleteClick: jest.fn(),
      onEditClick: jest.fn(),
    };

    component = shallow(<AuthorDetailView {...props} />);
  });

  test('renders correctly', () => {
    expect(component.find('.bookly-author-detail-view').length).toEqual(1);
    expect(component.find(Card).length).toEqual(1);
  });
});
