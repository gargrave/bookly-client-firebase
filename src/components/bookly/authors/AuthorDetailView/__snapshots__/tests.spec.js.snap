// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorDetailView matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AuthorDetailView
    author={Object {}}
    booksForAuthor={
      Array [
        Object {
          "author": Object {
            "bookCount": 1,
            "created": 2017-07-01T01:16:43.881Z,
            "firstName": "Aldous",
            "id": "oiw223h3qndlksadffl3ksdf",
            "lastName": "Huxley",
            "updated": 2017-07-01T01:16:43.881Z,
          },
          "created": 2018-01-14T01:36:26.220Z,
          "id": "sv3hmbefpc95MRXQsTHr",
          "title": "Brave New World",
          "updated": 2018-01-14T01:36:26.220Z,
        },
        Object {
          "author": Object {
            "bookCount": 1,
            "created": 2017-07-01T01:16:43.881Z,
            "firstName": "Aldous",
            "id": "oiw223h3qndlksadffl3ksdf",
            "lastName": "Huxley",
            "updated": 2017-07-01T01:16:43.881Z,
          },
          "created": 2018-01-15T01:36:26.220Z,
          "id": "Hmh9eR3rs5bMvsTXfpcQ",
          "title": "Island",
          "updated": 2018-01-15T01:36:26.220Z,
        },
        Object {
          "author": Object {
            "bookCount": 1,
            "created": 2017-07-01T01:16:43.881Z,
            "firstName": "Aldous",
            "id": "oiw223h3qndlksadffl3ksdf",
            "lastName": "Huxley",
            "updated": 2017-07-01T01:16:43.881Z,
          },
          "created": 2018-01-16T01:36:26.220Z,
          "id": "eRpcQ5H3rsTXfmh9bMvs",
          "title": "The Doors of Perception",
          "updated": 2018-01-16T01:36:26.220Z,
        },
        Object {
          "author": Object {
            "bookCount": 1,
            "created": 2017-07-06T22:02:11.504Z,
            "firstName": "Kurt",
            "id": "9328fhs938hfdsdblkhf98",
            "lastName": "Vonnegut",
            "updated": 2017-07-06T22:02:11.504Z,
          },
          "created": 2017-11-01T01:36:26.220Z,
          "id": "e2TedzRQJKJ1VL5m7QhG",
          "title": "Mother Night",
          "updated": 2017-11-01T01:36:26.220Z,
        },
        Object {
          "author": Object {
            "bookCount": 1,
            "created": 2017-07-06T22:02:11.504Z,
            "firstName": "Kurt",
            "id": "9328fhs938hfdsdblkhf98",
            "lastName": "Vonnegut",
            "updated": 2017-07-06T22:02:11.504Z,
          },
          "created": 2017-11-01T01:39:26.220Z,
          "id": "8ZYttrQY042IhlZ11BVJ",
          "title": "Breakfast of Champions",
          "updated": 2017-11-01T01:39:26.220Z,
        },
      ]
    }
    onBackClick={[Function]}
    onBookAddClick={[Function]}
    onBookClick={[Function]}
    onDeleteClick={[Function]}
    onEditClick={[Function]}
    topLevelError=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "",
        <AuthorDetailCard
          author={Object {}}
          onBackClick={[Function]}
          onDeleteClick={[Function]}
          onEditClick={[Function]}
        />,
        <hr />,
        <AuthorBooksList
          author={Object {}}
          books={
            Array [
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-01T01:16:43.881Z,
                  "firstName": "Aldous",
                  "id": "oiw223h3qndlksadffl3ksdf",
                  "lastName": "Huxley",
                  "updated": 2017-07-01T01:16:43.881Z,
                },
                "created": 2018-01-14T01:36:26.220Z,
                "id": "sv3hmbefpc95MRXQsTHr",
                "title": "Brave New World",
                "updated": 2018-01-14T01:36:26.220Z,
              },
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-01T01:16:43.881Z,
                  "firstName": "Aldous",
                  "id": "oiw223h3qndlksadffl3ksdf",
                  "lastName": "Huxley",
                  "updated": 2017-07-01T01:16:43.881Z,
                },
                "created": 2018-01-15T01:36:26.220Z,
                "id": "Hmh9eR3rs5bMvsTXfpcQ",
                "title": "Island",
                "updated": 2018-01-15T01:36:26.220Z,
              },
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-01T01:16:43.881Z,
                  "firstName": "Aldous",
                  "id": "oiw223h3qndlksadffl3ksdf",
                  "lastName": "Huxley",
                  "updated": 2017-07-01T01:16:43.881Z,
                },
                "created": 2018-01-16T01:36:26.220Z,
                "id": "eRpcQ5H3rsTXfmh9bMvs",
                "title": "The Doors of Perception",
                "updated": 2018-01-16T01:36:26.220Z,
              },
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-06T22:02:11.504Z,
                  "firstName": "Kurt",
                  "id": "9328fhs938hfdsdblkhf98",
                  "lastName": "Vonnegut",
                  "updated": 2017-07-06T22:02:11.504Z,
                },
                "created": 2017-11-01T01:36:26.220Z,
                "id": "e2TedzRQJKJ1VL5m7QhG",
                "title": "Mother Night",
                "updated": 2017-11-01T01:36:26.220Z,
              },
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-06T22:02:11.504Z,
                  "firstName": "Kurt",
                  "id": "9328fhs938hfdsdblkhf98",
                  "lastName": "Vonnegut",
                  "updated": 2017-07-06T22:02:11.504Z,
                },
                "created": 2017-11-01T01:39:26.220Z,
                "id": "8ZYttrQY042IhlZ11BVJ",
                "title": "Breakfast of Champions",
                "updated": 2017-11-01T01:39:26.220Z,
              },
            ]
          }
          onBookAddClick={[Function]}
          onBookClick={[Function]}
        />,
      ],
      "className": "bookly-detail-view bookly-author-detail-view",
    },
    "ref": null,
    "rendered": Array [
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "author": Object {},
          "onBackClick": [Function],
          "onDeleteClick": [Function],
          "onEditClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "author": Object {},
          "books": Array [
            Object {
              "author": Object {
                "bookCount": 1,
                "created": 2017-07-01T01:16:43.881Z,
                "firstName": "Aldous",
                "id": "oiw223h3qndlksadffl3ksdf",
                "lastName": "Huxley",
                "updated": 2017-07-01T01:16:43.881Z,
              },
              "created": 2018-01-14T01:36:26.220Z,
              "id": "sv3hmbefpc95MRXQsTHr",
              "title": "Brave New World",
              "updated": 2018-01-14T01:36:26.220Z,
            },
            Object {
              "author": Object {
                "bookCount": 1,
                "created": 2017-07-01T01:16:43.881Z,
                "firstName": "Aldous",
                "id": "oiw223h3qndlksadffl3ksdf",
                "lastName": "Huxley",
                "updated": 2017-07-01T01:16:43.881Z,
              },
              "created": 2018-01-15T01:36:26.220Z,
              "id": "Hmh9eR3rs5bMvsTXfpcQ",
              "title": "Island",
              "updated": 2018-01-15T01:36:26.220Z,
            },
            Object {
              "author": Object {
                "bookCount": 1,
                "created": 2017-07-01T01:16:43.881Z,
                "firstName": "Aldous",
                "id": "oiw223h3qndlksadffl3ksdf",
                "lastName": "Huxley",
                "updated": 2017-07-01T01:16:43.881Z,
              },
              "created": 2018-01-16T01:36:26.220Z,
              "id": "eRpcQ5H3rsTXfmh9bMvs",
              "title": "The Doors of Perception",
              "updated": 2018-01-16T01:36:26.220Z,
            },
            Object {
              "author": Object {
                "bookCount": 1,
                "created": 2017-07-06T22:02:11.504Z,
                "firstName": "Kurt",
                "id": "9328fhs938hfdsdblkhf98",
                "lastName": "Vonnegut",
                "updated": 2017-07-06T22:02:11.504Z,
              },
              "created": 2017-11-01T01:36:26.220Z,
              "id": "e2TedzRQJKJ1VL5m7QhG",
              "title": "Mother Night",
              "updated": 2017-11-01T01:36:26.220Z,
            },
            Object {
              "author": Object {
                "bookCount": 1,
                "created": 2017-07-06T22:02:11.504Z,
                "firstName": "Kurt",
                "id": "9328fhs938hfdsdblkhf98",
                "lastName": "Vonnegut",
                "updated": 2017-07-06T22:02:11.504Z,
              },
              "created": 2017-11-01T01:39:26.220Z,
              "id": "8ZYttrQY042IhlZ11BVJ",
              "title": "Breakfast of Champions",
              "updated": 2017-11-01T01:39:26.220Z,
            },
          ],
          "onBookAddClick": [Function],
          "onBookClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "",
          <AuthorDetailCard
            author={Object {}}
            onBackClick={[Function]}
            onDeleteClick={[Function]}
            onEditClick={[Function]}
          />,
          <hr />,
          <AuthorBooksList
            author={Object {}}
            books={
              Array [
                Object {
                  "author": Object {
                    "bookCount": 1,
                    "created": 2017-07-01T01:16:43.881Z,
                    "firstName": "Aldous",
                    "id": "oiw223h3qndlksadffl3ksdf",
                    "lastName": "Huxley",
                    "updated": 2017-07-01T01:16:43.881Z,
                  },
                  "created": 2018-01-14T01:36:26.220Z,
                  "id": "sv3hmbefpc95MRXQsTHr",
                  "title": "Brave New World",
                  "updated": 2018-01-14T01:36:26.220Z,
                },
                Object {
                  "author": Object {
                    "bookCount": 1,
                    "created": 2017-07-01T01:16:43.881Z,
                    "firstName": "Aldous",
                    "id": "oiw223h3qndlksadffl3ksdf",
                    "lastName": "Huxley",
                    "updated": 2017-07-01T01:16:43.881Z,
                  },
                  "created": 2018-01-15T01:36:26.220Z,
                  "id": "Hmh9eR3rs5bMvsTXfpcQ",
                  "title": "Island",
                  "updated": 2018-01-15T01:36:26.220Z,
                },
                Object {
                  "author": Object {
                    "bookCount": 1,
                    "created": 2017-07-01T01:16:43.881Z,
                    "firstName": "Aldous",
                    "id": "oiw223h3qndlksadffl3ksdf",
                    "lastName": "Huxley",
                    "updated": 2017-07-01T01:16:43.881Z,
                  },
                  "created": 2018-01-16T01:36:26.220Z,
                  "id": "eRpcQ5H3rsTXfmh9bMvs",
                  "title": "The Doors of Perception",
                  "updated": 2018-01-16T01:36:26.220Z,
                },
                Object {
                  "author": Object {
                    "bookCount": 1,
                    "created": 2017-07-06T22:02:11.504Z,
                    "firstName": "Kurt",
                    "id": "9328fhs938hfdsdblkhf98",
                    "lastName": "Vonnegut",
                    "updated": 2017-07-06T22:02:11.504Z,
                  },
                  "created": 2017-11-01T01:36:26.220Z,
                  "id": "e2TedzRQJKJ1VL5m7QhG",
                  "title": "Mother Night",
                  "updated": 2017-11-01T01:36:26.220Z,
                },
                Object {
                  "author": Object {
                    "bookCount": 1,
                    "created": 2017-07-06T22:02:11.504Z,
                    "firstName": "Kurt",
                    "id": "9328fhs938hfdsdblkhf98",
                    "lastName": "Vonnegut",
                    "updated": 2017-07-06T22:02:11.504Z,
                  },
                  "created": 2017-11-01T01:39:26.220Z,
                  "id": "8ZYttrQY042IhlZ11BVJ",
                  "title": "Breakfast of Champions",
                  "updated": 2017-11-01T01:39:26.220Z,
                },
              ]
            }
            onBookAddClick={[Function]}
            onBookClick={[Function]}
          />,
        ],
        "className": "bookly-detail-view bookly-author-detail-view",
      },
      "ref": null,
      "rendered": Array [
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "author": Object {},
            "onBackClick": [Function],
            "onDeleteClick": [Function],
            "onEditClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "author": Object {},
            "books": Array [
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-01T01:16:43.881Z,
                  "firstName": "Aldous",
                  "id": "oiw223h3qndlksadffl3ksdf",
                  "lastName": "Huxley",
                  "updated": 2017-07-01T01:16:43.881Z,
                },
                "created": 2018-01-14T01:36:26.220Z,
                "id": "sv3hmbefpc95MRXQsTHr",
                "title": "Brave New World",
                "updated": 2018-01-14T01:36:26.220Z,
              },
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-01T01:16:43.881Z,
                  "firstName": "Aldous",
                  "id": "oiw223h3qndlksadffl3ksdf",
                  "lastName": "Huxley",
                  "updated": 2017-07-01T01:16:43.881Z,
                },
                "created": 2018-01-15T01:36:26.220Z,
                "id": "Hmh9eR3rs5bMvsTXfpcQ",
                "title": "Island",
                "updated": 2018-01-15T01:36:26.220Z,
              },
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-01T01:16:43.881Z,
                  "firstName": "Aldous",
                  "id": "oiw223h3qndlksadffl3ksdf",
                  "lastName": "Huxley",
                  "updated": 2017-07-01T01:16:43.881Z,
                },
                "created": 2018-01-16T01:36:26.220Z,
                "id": "eRpcQ5H3rsTXfmh9bMvs",
                "title": "The Doors of Perception",
                "updated": 2018-01-16T01:36:26.220Z,
              },
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-06T22:02:11.504Z,
                  "firstName": "Kurt",
                  "id": "9328fhs938hfdsdblkhf98",
                  "lastName": "Vonnegut",
                  "updated": 2017-07-06T22:02:11.504Z,
                },
                "created": 2017-11-01T01:36:26.220Z,
                "id": "e2TedzRQJKJ1VL5m7QhG",
                "title": "Mother Night",
                "updated": 2017-11-01T01:36:26.220Z,
              },
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-06T22:02:11.504Z,
                  "firstName": "Kurt",
                  "id": "9328fhs938hfdsdblkhf98",
                  "lastName": "Vonnegut",
                  "updated": 2017-07-06T22:02:11.504Z,
                },
                "created": 2017-11-01T01:39:26.220Z,
                "id": "8ZYttrQY042IhlZ11BVJ",
                "title": "Breakfast of Champions",
                "updated": 2017-11-01T01:39:26.220Z,
              },
            ],
            "onBookAddClick": [Function],
            "onBookClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
